@app/server/src/plugins/CreateUploadUrlPlugin.ts

```diff
@@ -0,0 +1,137 @@
+import { awsRegion, uploadBucket } from "@app/config";
+import * as aws from "aws-sdk";
+import { gql, makeExtendSchemaPlugin } from "graphile-utils";
+import uuidv4 from "uuid/v4";
+
+import { OurGraphQLContext } from "../middleware/installPostGraphile";
+
+enum AllowedUploadContentType {
+  IMAGE_APNG = "image/apng",
+  IMAGE_BMP = "image/bmp",
+  IMAGE_GIF = "image/gif",
+  IMAGE_JPEG = "image/jpeg",
+  IMAGE_PNG = "image/png",
+  IMAGE_SVG_XML = "image/svg+xml",
+  IMAGE_TIFF = "image/tiff",
+  IMAGE_WEBP = "image/webp",
+}
+
+interface CreateUploadUrlInput {
+  clientMutationId?: string;
+  contentType: AllowedUploadContentType;
+}
+
+const CreateUploadUrlPlugin = makeExtendSchemaPlugin(() => ({
+  typeDefs: gql`
+    """
+    The set of content types that we allow users to upload.
+    """
+    enum AllowedUploadContentType {
+      "image/apng"
```

Comment:
For GraphQL documentation, this should use triple-quote syntax:

```suggestion
      """image/apng"""
```